src/v3/plugins/github/graphView.test.js

```diff
@@ -0,0 +1,383 @@
+          expect(comment).toMatchSnapshot();
+        });
+        it("has the right parent", () => {
+          expect(view.parent(comment)).toEqual(issue);
+        });
+        it("is authored by decentralion", () => {
+          expect(Array.from(view.authors(comment))).toEqual([decentralion]);
+        });
+      });
+    });
+  });
+
+  describe("pulls", () => {
+    const pulls = Array.from(view.pulls(repo));
+    it("number of pulls matches snapshot", () => {
+      expect(pulls.length).toMatchSnapshot();
+    });
+
+    describe("/#5", () => {
+      const pull = pulls[1];
+      it("matches snapshot", () => {
+        expect(pull).toMatchSnapshot();
+      });
+      it("is pull #5", () => {
+        expect(pull.number).toBe("5");
+      });
+      it("is authored by decentralion", () => {
+        expect(Array.from(view.authors(pull))).toEqual([decentralion]);
+      });
+      it("has the right parent", () => {
+        expect(view.parent(pull)).toEqual(repo);
+      });
+      const comments = Array.from(view.comments(pull));
+      it("number of comments matches snapshot", () => {
+        expect(comments.length).toMatchSnapshot();
+      });
+
+      describe("/comment #1", () => {
+        const comment = comments[0];
+        it("matches snapshot", () => {
+          expect(comment).toMatchSnapshot();
+        });
+        it("has the right parent", () => {
+          expect(view.parent(comment)).toEqual(pull);
+        });
+        it("is authored by wchargin", () => {
+          expect(Array.from(view.authors(comment))).toEqual([wchargin]);
+        });
+      });
+      const reviews = Array.from(view.reviews(pull));
+      it("number of reviews matches snapshot", () => {
+        expect(reviews.length).toMatchSnapshot();
+      });
+
+      describe("/review #1", () => {
+        const review = reviews[0];
+        it("matches snapshot", () => {
+          expect(review).toMatchSnapshot();
+        });
+        it("has the right parent", () => {
+          expect(view.parent(review)).toEqual(pull);
+        });
+        it("is authored by wchargin", () => {
+          expect(Array.from(view.authors(review))).toEqual([wchargin]);
+        });
+        const reviewComments = Array.from(view.comments(review));
+        it("has the right number of review comments", () => {
+          expect(reviewComments.length).toMatchSnapshot();
+        });
+
+        describe("/comment #1", () => {
+          const reviewComment = reviewComments[0];
+          it("is authored by wchargin", () => {
+            expect(Array.from(view.authors(reviewComment))).toEqual([wchargin]);
+          });
+          it("matches snapshot", () => {
+            expect(reviewComment).toMatchSnapshot();
+          });
+          it("has the right parent", () => {
+            expect(view.parent(reviewComment)).toEqual(review);
+          });
+        });
+      });
+    });
+  });
+  describe("invariants", () => {
+    const userlike: GN.UserlikeAddress = {
```

Comment:
I'm open to it, although not persuaded it would add much value.

If we wanted to change the internal format without affecting clients, we could also hold this structured representation as the API and make an internal-only representation that we translate into. I'm not sure what benefits taking the constructor approach has over the approach I described (other than being cleaner).